### ISSUE TITLE
CPU DoS when generating HTML for crafted Markdown documents.

### ISSUE THREAD

**Issue by peng-hui** [gohugoio/hugo#8187]
### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
Hugo Static Site Generator v0.80.0-792EF0F4 linux/amd64 BuildDate: 2020-12-31T13:37:58Z
</pre>

### Does this issue reproduce with the latest release?
Yes.

#### Summary
Hugo can generate HTML web pages given certain Markdown documents. When specially crafting the Markdown file can lead to Hugo execute for a much longer time. For example, I only put one Markdown file under the `content/` folder, and run `hugo -D`, it takes over one minute to finish (even could not finish). I claim this is a performance issue that negatively affects user experience and needs to be addressed. 

#### Environment 
`Linux pc 4.9.0-14-amd64 #1 SMP Debian 4.9.246-2 (2020-12-17) x86_64 GNU/Linux`
#### Reproduce
1. Prepare special Markdown files and put them into the `content` folder. Here I provide several examples.
```python
"[" * 50000 + "]" * 50000
```
```python
"[[" * 50000 + "xxxx"  + "]]" * 50000
```
```python
"a <![CDATA[" * 50000
```
```python
"a" + "<?" * 50000
```
```python
"a <!A " * 50000
```
The above snippets are in python. You can use that to generate strings and store them in Markdown files for testing.
2. Each time with one such Markdown file in a fresh hugo site. You can `hugo -D` to generate a draft or build the site.
3. Check the time. Or in the previous step, you just use `time hugo -D` in Linux environment. You can observe obviously longer time.

**Comment by bep** [comment]
I would appreciate if you could raise this issue at https://github.com/yuin/goldmark/issues

I assume you mean that one should enforce some kind of length limit for markdown links etc?

**Comment by peng-hui** [comment]
> I would appreciate if you could raise this issue at https://github.com/yuin/goldmark/issues

I do not know what exact Goldmark Hugo uses in the latest version. Could you help me if you have time? Otherwise, I will post there sometime later.

> I assume you mean that one should enforce some kind of length limit for markdown links etc?

Yeah, enforcing the length limit shall be the simplest solution.

**Comment by bep** [comment]
https://github.com/gohugoio/hugo/blob/master/go.mod#L59

**Comment by jmooring** [comment]
This was resolved in v0.93.3 with https://github.com/gohugoio/hugo/commit/970f385c48140e2b21028e6952896f324b48a76e
